homeassistant/components/zha/__init__.py

```diff
@@ -152,3 +146,19 @@
         await hass.config_entries.async_forward_entry_unload(config_entry, component)
 
     return True
+
+
+async def async_load_entities(
+    hass: HomeAssistantType, config_entry: config_entries.ConfigEntry
+) -> None:
+    """Load entities after integration was setup."""
+    to_setup = [
+        hass.data[DATA_ZHA][DATA_ZHA_PLATFORM_LOADED][comp]
+        for comp in COMPONENTS
+        if hass.data[DATA_ZHA][comp]
+    ]
+    results = await asyncio.gather(*to_setup, return_exceptions=True)
+    for res in results:
+        if isinstance(res, Exception):
+            _LOGGER.warning("Couldn't setup zha platform: %s", res)
```

Comment:
the platform is setup right? This should be an error creating a specific entity or collection of entities?

Reply:
Interesting, are you conditionally loading platforms like this? is that what `if hass.data[DATA_ZHA][comp]` is for? only load the platform if we have stuff for it?

Reply:
Well, that was just an extra precaution, if platform loading throws, wanted to log it, instead of getting `Exception was never retrieved...` unless `hass.create_task()` does that for us?